src/psyclone/parse/algorithm.py

```diff
@@ -201,14 +237,54 @@ def parse(self, alg_filename):
             # Nothing relevant found.
             raise ParseError(
                 "algorithm.py:parser:parse: Program, module, function or "
-                "subroutine not found in parse tree for file "
-                "'{0}'".format(alg_filename))
+                "subroutine not found in fparser2 parse tree.")
 
         self._unique_invoke_labels = []
         self._arg_name_to_module_name = OrderedDict()
+        # Holds variable name, type and precision information.
+        self._arg_type_defns = dict()
         invoke_calls = []
 
         for statement in walk(alg_parse_tree.content):
```

Comment:
You could specify `types` to this walk() but I'm not sure it would buy us very much?

Reply:
Good point. It could improve efficiency.